shopify-client/src/views/ShopList.vue

```diff
@@ -0,0 +1,261 @@
+          >
+        </div>
+      </div>
+      <div class="filterTile">
+        <h4>Tags</h4>
+        <div v-if="tags.length < 1">
+          <p>No available tags</p>
+        </div>
+        <div
+            v-else
+            v-for="tag in tags"
+            :key="tag.id"
+        >
+          <label :for="tag.name">{{tag.name}}</label>
+          <input
+              type="checkbox"
+              :id="tag.name"
+              class="checkBox"
+          >
+        </div>
+      </div>
+      <input
+          id="filterButton"
+          type="button"
+          value="Filter Search"
+          v-on:click="filterSearch"
+      />
+    </div>
+    <div id="filteredList">
+      <h4 id="searchKeyword">Search keyword: "{{keyword}}"</h4>
+      <div v-if="shops.length < 1">
+        No results found
+      </div>
+      <div
+          v-else
+          v-for="shop in shops"
+          :key="shop.id"
+          class="shopTile"
+      >
+        <img :src="'/' + shop.image"/> <!--Update when root image location is known -->
+        <h3>
+          <a :href="'/shop/' + shop.id">
+            {{shop.name.toLocaleUpperCase()}}
+          </a>
+        </h3>
+        <p>
+          {{shop.description}}
+        </p>
+        <div
+            v-for="category in shop.categories"
+            :key="category.id"
+            :class="'shopAttribute shopCategory ' + category.name"
+        >
+          {{category.name}}
+        </div>
+        <div
+            v-for="tag in shop.tags"
+            :key="tag.id"
+            :class="'shopAttribute shopTag ' + tag.name"
+        >
+          {{tag.name}}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+//Placeholders
+//import header from '@/components/header.vue'
+//import footer from '@/components/footer.vue'
+
+let addDiff = (newAttributes, target) =>
+    newAttributes.forEach(function (attribute){
+      if(target.every(element => attribute.name !== element.name)){
+        target.push(attribute)
+      }
+    });
+
+//checker if the shop has the selected tag and categories
+let checker = (shop, target) =>
+    target.every(attribute => shop.getElementsByClassName(attribute).length > 0);
+
+export default {
+  name: "stopList",
+  data(){
+    return{
+      shops:[],
+      categories: [],
+      tags: [],
+      keyword: '',
+    }
+  },
+  created: function(){
+    const vm = this;
+    vm.keyword = new URLSearchParams(window.location.search).get('keyword');
+
+    if(vm.keyword === null){
+      vm.keyword = ''
+    }
+
+    fetch('/api/shops/shopList?keyword=' + vm.keyword)
+        .then((response) => response.json())
+        .then((response) => {
+          if (response.status == 404) {
+            throw new Error(response.message);
+          }
+          return response;
+        })
+        .then((data) => {
+          vm.shops = data;
+
+          data.forEach(function(shop){
+            addDiff(shop.categories, vm.categories);
```

Comment:
You should send separate requests to retrieve all the categories and tags. I think `fetch` returns a Promise. Have a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
I do not want all categories and tags, just the ones that are available for the resulting shops

Reply:
The shops view should display all shops right? Users should be able to see all the available tags and categories.

Reply:
the shop list contains all shops, or ones that was affected by the searchbar

Reply:
The search bar is only a type of filter.

Reply:
I guess it kinda makes sense to only show tags and categories that are being used to avoid clutters.